vaadin-crud-flow-parent/vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -190,25 +190,20 @@ public Crud() {
         deleteButton.getElement().setAttribute("slot", "delete-button");
         deleteButton.addThemeNames("tertiary", "error");
         getElement().appendChild(deleteButton.getElement());
-
-        addAttachListener(e -> {
-            getElement()
-                    .executeJs("this.__validate = function () {return true;}");
-            // Override onFormChanges to dispatch an event so the server can
-            // control the dirty state of the form
-            getElement().executeJs(
-                    "this.__onFormChanges = function () { this.dispatchEvent(new CustomEvent('form-change')); }");
-        });
-
-        getElement().addEventListener("form-change", this::formChangeEvent);
     }
 
-    private void formChangeEvent(com.vaadin.flow.dom.DomEvent e) {
-        if (this.saveButton.isEnabled()) {
-            this.setDirty(true);
+    private class SaveButton extends Button {
+        @Override
+        public void onEnabledStateChanged(boolean enabled) {
+            super.onEnabledStateChanged(enabled);
+            overrideSaveDisabled(enabled);
         }
     }
 
+    private void overrideSaveDisabled(boolean enabled) {
+        getElement().executeJs("this.__isSaveBtnDisabled = () => {return $0;}", !enabled);
```

Comment:
This change will be undone when detaching + reattaching the component. I think it would need to be triggered again in onAttach.

Reply:
Isn't the only case that this part gets executed is when `onEnabledStateChanged` is called? I think it is independent of attach/detaching crud element to the DOM.

Reply:
`executeJs` calls do not persist over detaching and reattaching, they have to be called again after reattaching the component.

So for the following case:
- set button to disabled
- detach Crud
- reattach Crud

the `__isSaveBtnDisabled` override will not be in place anymore.

Reply:
Good point! The logic is now in place.